modules/heat_conduction/src/meshgenerators/PatchSidesetGenerator.C

```diff
@@ -91,6 +91,15 @@ PatchSidesetGenerator::generate()
   // get a list of all sides; vector of tuples (elem, loc_side, side_set)
   auto side_list = boundary_info.build_active_side_list();
 
+  // check if the provided sideset name is actually a sideset id
+  // if _sideset_name can be converted to integer it's interpreted
+  // as sideset id
+  std::stringstream ss;
+  ss << _sideset_name;
+  ss >> _sideset;
+  if (ss.fail())
+    _sideset = boundary_info.get_id_by_name(_sideset_name);
```

Comment:
`get_id_by_name()` doesn't error if the sideset isn't found - need to add a graceful exit here:

```suggestion
    _sideset = boundary_info.get_id_by_name(_sideset_name);
    if (_sideset == BoundaryInfo::invalid_id)
      paramError("sideset_name", "Not a valid boundary");
```